mautic/mautic
# Fix unique vs returning visitors widget

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
Unique vs returning visitors dashboard widget return wrong data. This PR try rebuild queryto return expected data

Query before

`SELECT COUNT(t.id) AS count FROM page_hits t WHERE t.date_hit BETWEEN :dateFrom AND :dateTo`

`SELECT COUNT(t.id) AS count FROM page_hits t WHERE (lead_id IS NULL) AND (t.date_hit BETWEEN :dateFrom AND :dateTo)`

Query after

`SELECT COUNT(*) as count FROM (SELECT t.id FROM page_hits t WHERE (lead_id IS NOT NULL) AND (date_left IS NULL) AND (redirect_id IS NULL)  AND (email_id IS NULL) AND (t.date_hit BETWEEN :dateFrom AND :dateTo) GROUP BY t.lead_id) tt`

`SELECT COUNT(*) as count FROM (SELECT t.id FROM page_hits t WHERE (lead_id IS NOT NULL) AND (date_left IS NULL) AND (redirect_id IS NULL)  AND (email_id IS NULL) AND (t.date_hit BETWEEN :dateFrom AND :dateTo) GROUP BY t.lead_id  HAVING COUNT(t.id) > 1) tt`




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create unique vs returning visitors dashboard widget on your instances with page_hits data
3. Check If you see correct pie graph 




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10302"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
